pkg/generate/app/sourcelookup.go

```diff
@@ -265,6 +269,26 @@ func (r *SourceRepository) AddDockerfile(contents string) error {
 	return nil
 }
 
+// AddBuildSecrets adds the defined secrets into a build. The input format for
+// the secrets is "<secretName>:<destinationDir>". The destinationDir is
+// optional and when not specified the default is the current working directory.
+func (r *SourceRepository) AddBuildSecrets(secrets []string) error {
```

Comment:
@csrwng @bparees I reworked this a little bit so the NewBuildPipeline does not need to pass `[]string` anymore. However, I still struggle if adding secrets to a source will work if you have no source or if that is possible to have with `new-build` command.